.github/workflows/wheels-build.yaml

```diff
@@ -241,6 +237,7 @@ jobs:
       - uses: actions/upload-artifact@v4
         if: ${{ inputs.upload-artifacts }}
         with:
+          if-no-files-found: 'error'
```

Comment:
Reference for this: https://github.com/actions/upload-artifact?tab=readme-ov-file#inputs

When `actions/upload-artifact` does not find any files to upload at `{path}`, by default it *warns* but does not *error*.

I think for our purposes, we want a big loud CI-failing error, as 0 files being found likely points to a misconfiguration. For any purposes where this workflow is run and you don't want to upload artifacts, we already have an opt-out via the `upload-artifacts` boolean input: https://github.com/rapidsai/shared-workflows/pull/340